content/blog/monitor-application-performance-across-hybrid-cloud-environment-using-apache-skywalking-and-service-mesh.md

```diff
@@ -0,0 +1,320 @@
+      CA_ADDR:                       istiod.istio-system.svc:15012
+      POD_NAME:                      productcatalogservice-77589df479-p2p2c (v1:metadata.name)
+      POD_NAMESPACE:                 boutique (v1:metadata.namespace)
+      INSTANCE_IP:                    (v1:status.podIP)
+      SERVICE_ACCOUNT:                (v1:spec.serviceAccountName)
+      HOST_IP:                        (v1:status.hostIP)
+      PROXY_CONFIG:                  {"envoyAccessLogService":{"address":"afd4a163a65c74af5ad732fcf86b7dff-261027448.us-east-2.elb.amazonaws.com:11800"},"envoyMetricsService":{"address":"afd4a163a65c74af5ad732fcf86b7dff-261027448.us-east-2.elb.amazonaws.com:11800"},"proxyStatsMatcher":{"inclusionRegexps":[".*"]}}
+                                     
+      ISTIO_META_POD_PORTS:          [
+                                         {"containerPort":3550,"protocol":"TCP"}
+                                     ]
+      ISTIO_META_APP_CONTAINERS:     server
+      ISTIO_META_CLUSTER_ID:         Kubernetes
+      ISTIO_META_INTERCEPTION_MODE:  REDIRECT
+      ISTIO_META_WORKLOAD_NAME:      productcatalogservice
+      ISTIO_META_OWNER:              kubernetes://apis/apps/v1/namespaces/boutique/deployments/productcatalogservice
+      ISTIO_META_MESH_ID:            cluster.local
+      TRUST_DOMAIN:                  cluster.local```
+````
+
+It should be noted that, in order to avoid collecting ALS logs for the Skupper deployments, e.g., *skupper-router* & *skupper-service-controller* deployed to the same namespace that are used for multicloud communication and hybrid application deployment, we can add the annotation *sidecar.istio.io/inject: false* to those _Skupper_ deployments. After restart deployments, the _Skupper_ deployments will not inject the *istio-proxy* to their pods. 
+
+#### 3 Monitor Online Boutique application
+
+After follow up all the steps, the Online Boutique application metrics will be showing up from Apache SkyWalking UI, under *Service Mesh* tab:
+
+![](/img/sw-app-svc.png)
+
+The SkyWalking UI *Topology* page will show the Online Boutique application topology map: 
+
+![](/img/sw-app-map.png)
+
+You can check the service *Overview* and *Endpoint* pages per service, e.g., *frontend*:
+
+![](/img/sw-app-overview.png)
+
+![](/img/sw-app-endpoint.png)
+
+You can also check all the services from the *Data Plane* tab under  *Service Mesh* that are observed through _Envoy Metrics Service_ setup:
+
+![](/img/sw-app-svc-dataplane.png)
+
+### Online Boutique application alerting
+
+
+
+Apache SkyWalking provides an alerting mechanism to measure application performance according to a list of pre-defined metrics, e.g., _service_resp_time_, _ service_instance_resp_time_, and _service_sla_. It will trigger alerting when some metrics reach its pre-defined thresholds. We can define new metrics using SkyWalking observability analysis language (OAL), or customize the existing metrics with new thresholds. 
```

Comment:
Please make sure to use escape character (a backslash) in front of "_" character for _service_resp_time and other metrics.